react/features/e2ee/middleware.js

```diff
@@ -35,6 +46,69 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
         dispatch(unregisterSound(E2EE_ON_SOUND_ID));
         break;
 
+    case PARTICIPANT_UPDATED:
+    case PARTICIPANT_JOINED: {
+        const result = next(action);
+        const { e2eeEnabled, e2eeSupported, local } = action.participant;
+        const { everyoneEnabledE2EE } = getState()['features/e2ee'];
+
+        // if all had it enabled and this one disabled it, change value in store
+        // otherwise there is no change in the value we store
+        if (everyoneEnabledE2EE && !e2eeEnabled) {
+            dispatch({
+                type: SET_EVERYONE_ENABLED,
+                everyoneEnabledE2EE: false
+            });
+        }
+
+        if (local) {
+            return result;
+        }
+
+        const { everyoneSupportsE2EE } = getState()['features/e2ee'];
+
+        // if all supported it and this one does not, change value in store
+        // otherwise there is no change in the value we store
+        if (everyoneSupportsE2EE && !e2eeSupported) {
+            dispatch({
+                type: SET_EVERYONE_SUPPORTS,
+                everyoneSupportsE2EE: false
+            });
+        }
+
+        return result;
+    }
+
+    case PARTICIPANT_LEFT: {
+        const result = next(action);
+        const { e2eeEnabled, e2eeSupported } = action.participant;
+
+        const { everyoneEnabledE2EE } = getState()['features/e2ee'];
+
+        const participants = getParticipants(getState());
+
+        // if it was not enabled by everyone, and the participant leaving had it disabled
+        // let's check is it enabled for all that stay
+        if (!everyoneEnabledE2EE && !e2eeEnabled && participants.every(p => p.e2eeEnabled)) {
```

Comment:
Maybe you can combine the loop over the participants for enable with the one for support?